File: app/src/main/java/com/kota/Bahamut/Dialogs/Dialog_PostArticle.java
package com.kota.Bahamut.Dialogs;

import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.RadioGroup;
import android.widget.Spinner;

import com.kota.ASFramework.Dialog.ASDialog;
import com.kota.Bahamut.R;

import java.util.Objects;

public class Dialog_PostArticle extends ASDialog implements View.OnClickListener {
    public static final int NEW = 0;
    public static final int REPLY = 1;
    Button _cancel_button;
    Dialog_PostArticle_Listener _listener = null;
    RadioGroup _post_target_group;
    Button _send_button;
    Spinner _sign_spinner;
    int _target;

    public String getName() {
        return "BahamutBoardsPostArticle";
    }

    public Dialog_PostArticle(int aTarget) {
        requestWindowFeature(1);
        setContentView(R.layout.dialog_post_article);
        Objects.requireNonNull(getWindow()).setBackgroundDrawable(null);
        this._target = aTarget;
        this._post_target_group = findViewById(R.id.post_target);
        this._send_button = findViewById(R.id.send);
        this._cancel_button = findViewById(R.id.cancel);
        View reply_target_view = findViewById(R.id.reply_target_view);
        findViewById(R.id.sign_view);
        if (this._target == 0) {
            reply_target_view.setVisibility(View.GONE);
        } else {
            reply_target_view.setVisibility(View.VISIBLE);
        }
        this._sign_spinner = findViewById(R.id.sign_spinner);
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getContext(), R.array.reply_target_list, R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        this._sign_spinner.setAdapter(adapter);
        this._send_button.setOnClickListener(this);
        this._cancel_button.setOnClickListener(this);
    }

    public void onClick(View view) {
        String target;
        if (view == this._send_button && this._listener != null) {
            int checked_id = this._post_target_group.getCheckedRadioButtonId();
            int selected_sign = this._sign_spinner.getSelectedItemPosition();
            String sign = "";
            if (selected_sign > 0) {
                sign = String.valueOf(selected_sign - 1);
            }
            if (checked_id == R.id.post_to_mail) {
                target = "M";
            } else if (checked_id == R.id.post_to_both) {
                target = "B";
            } else {
                target = "F";
            }
            this._listener.onPostArticleDoneWithTarger(target, sign);
        }
        dismiss();
    }

    public void setListener(Dialog_PostArticle_Listener listener) {
        this._listener = listener;
    }
}

File: app/src/main/java/com/kota/Bahamut/dialogs/DialogPostArticle.kt
package com.kota.Bahamut.dialogs

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.RadioButton
import androidx.compose.material3.RadioButtonDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import com.kota.Bahamut.R
import com.kota.Bahamut.service.CommonFunctions
import com.kota.Bahamut.ui.components.BahaText
import com.kota.Bahamut.ui.components.BahaTextSize
import com.kota.Bahamut.ui.dialogs.BahaAlertDialogContent
import com.kota.Bahamut.ui.dialogs.BahaDialogButton
import com.kota.Bahamut.ui.theme.AppTheme
import com.kota.asFramework.dialog.ASDialog
import com.kota.telnet.TelnetArticle

class DialogPostArticle(private val myTarget: Int) : ASDialog() {
    var dialogPostArticleListener: DialogPostArticleListener? = null

    override val name: String?
        get() = "BahamutBoardsPostArticle"

    init {
        setTitle(CommonFunctions.getContextString(R.string.confirm))
        setComposeContent {
            Content()
        }
    }

    @Composable
    private fun Content() {
        val context = LocalContext.current
        val colors = AppTheme.colors
        val signList = remember {
            context.resources.getStringArray(R.array.reply_target_list)
        }

        var selectedTarget by remember { mutableStateOf("F") } // "F" = Board, "M" = Mail, "B" = Both
        var selectedSignIndex by remember { mutableIntStateOf(0) }
        var isSignDropdownExpanded by remember { mutableStateOf(false) }

        val scrollState = rememberScrollState()

        BahaAlertDialogContent(
            modifier = Modifier.widthIn(min = 280.dp, max = 340.dp),
            title = stringResource(R.string.confirm),
            buttons = listOf(
                BahaDialogButton(
                    text = stringResource(R.string.cancel),
                    onClick = { dismiss() }
                ),
                BahaDialogButton(
                    text = stringResource(R.string.send),
                    onClick = {
                        val sign = if (selectedSignIndex > 0) (selectedSignIndex - 1).toString() else ""
                        dialogPostArticleListener?.onPostArticleDoneWithTarget(selectedTarget, sign)
                        dismiss()
                    }
                )
            )
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .verticalScroll(scrollState)
            ) {
                BahaText(
                    text = stringResource(R.string.is_post_article),
                    color = colors.textPrimary,
                    modifier = Modifier.padding(bottom = 6.dp),
                    size = BahaTextSize.TITLE
                )

                if (myTarget != TelnetArticle.NEW) {
                    Spacer(modifier = Modifier.height(8.dp))
                    BahaText(
                        text = stringResource(R.string.is_post_article_reply),
                        color = colors.textPrimary,
                        modifier = Modifier.padding(bottom = 2.dp)
                    )

                    val targets = listOf(
                        "F" to stringResource(R.string.post_to_board),
                        "M" to stringResource(R.string.post_to_mail),
                        "B" to stringResource(R.string.post_to_both)
                    )

                    targets.forEach { (key, label) ->
                        Box(
                            modifier = Modifier
                                .fillMaxWidth()
                                .height(48.dp)
                                .clickable { selectedTarget = key }
                                .padding(horizontal = 4.dp)
                        ) {
                            RadioButton(
                                selected = (selectedTarget == key),
                                onClick = { selectedTarget = key },
                                colors = RadioButtonDefaults.colors(
                                    selectedColor = colors.checkboxTint,
                                    unselectedColor = colors.textSecondary
                                ),
                                modifier = Modifier.align(Alignment.CenterStart)
                            )
                            BahaText(
                                text = label,
                                color = colors.textPrimary,
                                modifier = Modifier.align(Alignment.Center),
                                size = BahaTextSize.TITLE
                            )
                        }
                    }
                }

                Spacer(modifier = Modifier.height(10.dp))
                BahaText(
                    text = stringResource(R.string.select_sign),
                    color = colors.textPrimary,
                    modifier = Modifier.padding(bottom = 4.dp)
                )

                Box(modifier = Modifier.fillMaxWidth()) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(48.dp)
                            .clickable { isSignDropdownExpanded = true }
                            .padding(horizontal = 4.dp),
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        BahaText(
                            text = signList.getOrElse(selectedSignIndex) { "" },
                            color = colors.textPrimary,
                            size = BahaTextSize.TITLE
                        )
                        BahaText(
                            text = "▼",
                            color = colors.textPrimary,
                            modifier = Modifier.padding(end = 4.dp),
                            size = BahaTextSize.TITLE
                        )
                    }

                    DropdownMenu(
                        expanded = isSignDropdownExpanded,
                        onDismissRequest = { isSignDropdownExpanded = false },
                        modifier = Modifier.background(colors.pageBackground)
                    ) {
                        signList.forEachIndexed { index, name ->
                            DropdownMenuItem(
                                text = {
                                    BahaText(
                                        text = name,
                                        color = if (selectedSignIndex == index) colors.checkboxTint else colors.textPrimary
                                    )
                                },
                                onClick = {
                                    selectedSignIndex = index
                                    isSignDropdownExpanded = false
                                }
                            )
                        }
                    }
                }
            }
        }
    }

    fun setListener(listener: DialogPostArticleListener?) {
        this.dialogPostArticleListener = listener
    }
}
